tests/integration/tests/test_cloudbeat_integration.py

```diff
@@ -94,5 +72,37 @@ def test_elastic_index_exists(elastic_client, match_type):
             break
         time.sleep(1)
 
-    assert len(result) > 0,\
+    assert len(result) > 0, \
         f"The findings of type {match_type} not found"
+
+
+@pytest.mark.pre_merge
+@pytest.mark.order(4)
+@pytest.mark.dependency(depends=["test_cloudbeat_pod_exist"])
+def test_leader_election(fixture_data, elastic_client, cloudbeat_agent, k8s):
+    """
+    This test verifies that k8s related findings are sent by a single agent
+    :param fixture_data: (Pods list, Nodes list)
+    :param elastic_client: Elastic API client
+    :param cloudbeat_agent: Cloudbeat configuration
+    :param k8s: Kubernetes client object
+    :return:
+    """
+
+    query, sort = elastic_client.build_es_query(match_type="k8s_object")
+    pods, nodes = fixture_data
+    leader_node = k8s.get_cluster_leader(namespace=cloudbeat_agent.namespace, pods=pods)
+    assert leader_node != "", \
+        "The Leader node could not be found"
+
+    time.sleep(CONFIG_TIMEOUT)  # Wait for all pods to send resources to ES
+
+    result = elastic_client.get_index_data(index_name=elastic_client.index,
+                                           query=query,
+                                           size=1000,
+                                           sort=sort)
+    # checking that k8s_objects are being sent only by the leader node.
+    for resource in result['hits']['hits']:
+        assert leader_node == resource['_source']['agent']['name'], \
+            f"Multiple agents send k8s_objects, leader: {leader_node}, resource: {resource['_source']}"
```

Comment:
nit, failure message will display very long string with resource['_source'], because the last one returns big dictionary. Did you tried to simulate failure and check how it seems in the report? 
The simplest way to do that is set leader_node = "some value". Please verify that the message is readable.
if not you can  print resource['_source']['agent']['name'] and then to attach resource to allure report ( example can be found in test_elastic_index_exists test case).

Reply:
TY @gurevichdmitry, the message is readable.